htdocs/core/class/commonobject.class.php

```diff
@@ -5380,16 +5380,19 @@ protected function commonGenerateDocument($modelspath, $modele, $outputlangs, $h
 					return 1;
 				} else {
 					$outputlangs->charset_output = $sav_charset_output;
-					dol_print_error($this->db, "Error generating document for ".__CLASS__.". Error: ".$obj->error, $obj->errors);
+					dol_syslog("Error generating document for ".__CLASS__.". Error: ".$obj->error, LOG_ERR);
+					setEventMessages($obj->error, $obj->errors, 'errors');
```

Comment:
The method commonGenerateDocument is not a method into a GUI context (not a page). It is also used in batch context.
So the setEventMessages must not be used here. Can you try to replace with 
```
$this->error = $obj->error
$this->errors = $obj->errors

```

Reply:
@eldy 
You're right.
I replaced it.